CactusTutorial.ipynb

```diff
@@ -18,38 +18,70 @@
     "\n",
     "On Centos use this command:\n",
     "```bash\n",
-    "sudo yum install -y epel-release\n",
-    "sudo yum install -y libjpeg-turbo-devel gcc git lapack-devel make subversion gcc-c++ which papi-devel hwloc-devel openmpi-devel hdf5-openmpi-devel openssl-devel libtool-ltdl-devel numactl-devel gcc-gfortran hdf5-devel fftw-devel patch gsl-devel\n",
+    "su -c 'yum install -y epel-release'\n",
+    "su -c 'yum install -y libjpeg-turbo-devel gcc git lapack-devel make subversion gcc-c++ which papi-devel hwloc-devel openmpi-devel hdf5-openmpi-devel openssl-devel libtool-ltdl-devel numactl-devel gcc-gfortran hdf5-devel fftw-devel patch gsl-devel'\n",
+    "module load mpi/openmpi-x86_64\n",
     "```\n",
+    "You will have to repeat the `module load` command once in each new shell each time you would like to compile or run the code. You may have to log out and back in for the module command to become visible.\n",
     "\n",
     "On OpenSuse use this command:\n",
     "```bash\n",
     "sudo zypper install -y curl gcc git lapack-devel make subversion gcc-c++ which papi-devel hwloc-devel openmpi-devel libopenssl-devel libnuma-devel gcc-fortran hdf5-devel libfftw3-3 patch gsl-devel pkg-config\n",
-    "```"
+    "sudo mpi-selector --set  $(mpi-selector --list | head -n1)\n",
+    "```\n",
+    "You will only have to execute the `mpi-selector` once, after that log out and back in to make the `mpirun` and `mpicc` commands visible without which Cactus will compile very slowly and fail to run."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Notebook setup\n",
+    "This notebook is intended to be used online on the Einstein Toolkit tutorial server, offline as a read-only document, as a jupyter notebook that you can download and also in your own docker container using `ndslabs/jupyter-et`. To make all of these work some setting need to be tweaked, which we do in the next cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# this allows you to use \"cd\" in cells to change directories instead of requiring \"%cd\"\n",
+    "%automagic on\n",
+    "# override IPython's default %%bash to not buffer all output\n",
+    "from IPython.core.magic import register_cell_magic\n",
+    "@register_cell_magic\n",
+    "def bash(line, cell): get_ipython().system(cell)\n",
+    "# OpenMPI in docker containers produces warnings due to the vader library, this disables them\n",
+    "import os\n",
+    "os.environ[\"OMPI_MCA_btl_vader_single_copy_mechanism\"] = \"none\"\n",
+    "# Some versions of OpenMPI prevent oversubscribing cpus, which may happen if simfactory's\n",
+    "# number of cores detection is imperfect\n",
+    "os.environ[\"OMPI_MCA_rmaps_base_oversubscribe\"] = \"true\""
```

Comment:
I'm not thrilled with the setting of these obscure environment variables

Reply:
The `OMPI_MCA_btl_vader_single_copy_mechanism` can go as it seem to only occur in containers, I would rather keep `OMPI_MCA_rmaps_base_oversubscribe` as the `--procs 2` TOV simulation will otherwise fail with OpenMPI (ie if run on a system like Ubuntu where OpenMPI is the default MPI stack).

Reply:
I have removed the `OMPI_MCA_btl_vader_single_copy_mechanism`setting in a87bb49